test/ProgressOnderwijsUtils.Tests/UriExtensionsTest.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using ExpressionToCodeLib;
+using Xunit;
+
+namespace ProgressOnderwijsUtils.Tests
+{
+    public sealed class UriExtensionsTest
+    {
+        [Fact]
+        public void Combine_works_savely()
+        {
+            PAssert.That(() => new Uri("c:/dir").Combine("file").AbsolutePath == "c:/file");
+            PAssert.That(() => new Uri("c:/dir/").Combine("file").AbsolutePath == "c:/dir/file");
+            PAssert.That(() => new Uri("c:/dir").Combine("/file").AbsolutePath == "c:/file");
+            PAssert.That(() => new Uri("c:/dir/").Combine("/file").AbsolutePath == "c:/file");
+        }
+
+        [Fact]
+        public void RefersToDirectory_takes_leading_slash_into_account()
+        {
+            PAssert.That(() => !new Uri("c:/file").RefersToDirectory());
+            PAssert.That(() => new Uri("c:/dir/").RefersToDirectory());
+        }
+
+        [Fact]
+        public void EnsureRefersToDirectory_adds_needed_traling_slash()
+        {
+            PAssert.That(() => new Uri("c:/file").EnsureRefersToDirectory().RefersToDirectory());
+            PAssert.That(() => new Uri("c:/dir/").EnsureRefersToDirectory().RefersToDirectory());
+        }
+
+        [Fact]
+        public void ReplaceRootWith()
+        {
+            PAssert.That(() => new Uri("c:/file").ReplaceRootWith(new Uri("d:/")).AbsolutePath == "d:/file");
+            PAssert.That(() => new Uri("c:/dir/").ReplaceRootWith(new Uri("d:/")).AbsolutePath == "d:/dir/");

```

Comment:
En hier willen we zeker ook eentje met een SMB pad, want dat gebruiken we ook echt.